metrics_api/lib/opentelemetry/metrics/meter.rb

```diff
@@ -8,71 +8,156 @@ module OpenTelemetry
+      #   Must conform to the instrument name syntax:
+      #   not nil or empty, case-insensitive ASCII string that matches `/\A[a-z][a-z0-9_.-]{0,62}\z/i`
+      # @param unit [optional String]
+      #   Must conform to the instrument unit rule:
+      #   case-sensitive ASCII string with maximum length of 63 characters
+      # @param description [optional String]
+      #   Must conform to the instrument description rule:
+      #   UTF-8 string but up to 3 bytes per charater with maximum length of 1023 characters
+      # @param callback [optional Proc, Array<Proc>]
+      #   Callback functions should:
+      #   - be reentrant safe;
+      #   - not take an indefinite amount of time;
+      #   - not make duplicate observations (more than one Measurement with the same attributes)
+      #     across all registered callbacks;
+      #
+      # @return [Instrument::ObservableGauge]
+      def create_observable_gauge(name, unit: nil, description: nil, callback: nil)
+        create_instrument(:observable_gauge, name, unit, description, callback) { NOOP_OBSERVABLE_GAUGE }
       end
 
-      def create_observable_up_down_counter(name, unit: nil, description: nil, callback:)
-        create_instrument(:observable_up_down_counter, name, unit, description, callback) { OBSERVABLE_UP_DOWN_COUNTER }
+      # @param name [String]
+      #   Must conform to the instrument name syntax:
+      #   not nil or empty, case-insensitive ASCII string that matches `/\A[a-z][a-z0-9_.-]{0,62}\z/i`
+      # @param unit [optional String]
+      #   Must conform to the instrument unit rule:
+      #   case-sensitive ASCII string with maximum length of 63 characters
+      # @param description [optional String]
+      #   Must conform to the instrument description rule:
+      #   UTF-8 string but up to 3 bytes per charater with maximum length of 1023 characters
+      # @param callback [optional Proc, Array<Proc>]
+      #   Callback functions should:
+      #   - be reentrant safe;
+      #   - not take an indefinite amount of time;
+      #   - not make duplicate observations (more than one Measurement with the same attributes)
+      #     across all registered callbacks;
+      #
+      # @return [Instrument::ObservableUpDownCounter]
+      def create_observable_up_down_counter(name, unit: nil, description: nil, callback: nil)
+        create_instrument(:observable_up_down_counter, name, unit, description, callback) { NOOP_OBSERVABLE_UP_DOWN_COUNTER }
       end
 
       private
 
       def create_instrument(kind, name, unit, description, callback)
-        raise InstrumentNameError if name.nil?
-        raise InstrumentNameError if name.empty?
-        raise InstrumentNameError unless NAME_REGEX.match?(name)
-        raise InstrumentUnitError if unit && (!unit.ascii_only? || unit.size > 63)
-        raise InstrumentDescriptionError if description && (description.size > 1023 || !utf8mb3_encoding?(description.dup))
+        name = name.downcase
```

Comment:
This is new. From the Metrics' Meter API specification for the [Instrument Name Syntax](https://opentelemetry.io/docs/reference/specification/metrics/api/#instrument-name-syntax) it says:

> They are case-insensitive, ASCII strings.

So here I'm normalizing the name with `downcase` before using it in `@instrument_registry[name]`